pkg/volume/empty_dir/empty_dir.go

```diff
@@ -66,27 +66,29 @@ func (plugin *emptyDirPlugin) Name() string {
 	return emptyDirPluginName
 }
 
-func (plugin *emptyDirPlugin) CanSupport(spec *api.Volume) bool {
+func (plugin *emptyDirPlugin) CanSupport(spec *volume.InternalVolume) bool {
 	if plugin.legacyMode {
 		// Legacy mode instances can be cleaned up but not created anew.
 		return false
 	}
 
-	if util.AllPtrFieldsNil(&spec.VolumeSource) {
+	if util.AllPtrFieldsNil(spec.PersistentVolumeSource) ||
```

Comment:
EmptyDir is explicitly the default if AllPtrFieldsNil(VS) - that is intended.  Ahh, but that is patched up in validation, which means this whole check can probably be removed!  Yay.